modules/history/src/history-helper.js

```diff
@@ -0,0 +1,173 @@
+const SerializableClass = require("abitbol-serializable");
+const self = require("../index.js");
+
+const historyHelper = {
+
+    /**
+     * Check the differences between two state and apply the changes of snapshotState
+     * to currentState
+     * @param  {Object} currentState - The current state of the history
+     * @param  {Object} snapshotState - The new state of the history
+     */
+    applySnapshotDifference: (currentState, snapshotState) => { // eslint-disable-line
+        const {
+            dataStore,
+        } = self.app.modules;
+        // We make a flat object list for each object
+        const currentEntities = historyHelper.flattenSnapshot(currentState);
+        const snapshotEntities = historyHelper.flattenSnapshot(snapshotState);
+
+        Object.keys(currentEntities).forEach(
+            (currentEntityKey) => {
+                const snapshotEntitiesKeys = Object.keys(snapshotEntities);
+                const snapshotEntityKey = snapshotEntitiesKeys.find(
+                    snapEntityKey => (currentEntityKey === snapEntityKey),
+                );
+                if (typeof snapshotEntityKey !== "undefined") {
+                    // Check common entity properties
+                    const currentEntity = currentEntities[currentEntityKey];
+                    const snapshotEntity = snapshotEntities[snapshotEntityKey];
+                    const entity = dataStore.getEntity(currentEntityKey);
+                    // Property is different or doesn't exist
+                    const unserializedEntity = SerializableClass
+                        .$unserialize(snapshotEntity);
+                    Object.keys(snapshotEntity).forEach(
+                        (property) => {
+                            if ((typeof currentEntity[property] !== "undefined"
+                                && historyHelper.checkDiff(currentEntity[property],
+                                    snapshotEntity[property]))) {
+                                // Found differences
+                                entity[property] = unserializedEntity[property];
```

Comment:
As we said earlier, a property can't be "deleted" between two states or added as it show inconsistency. Also setting a property to "undefined" can't be done because it can't be serialized so this case also should not happen. But maybe a warning is necessary